zap/src/main/java/org/parosproxy/paros/core/scanner/Alert.java

```diff
@@ -1277,10 +1284,32 @@ public final Alert build() {
             if (alertRef != null) {
                 alert.setAlertRef(alertRef);
             }
-            alert.setTags(tags);
+            addCweTag(cweId);
+            alert.setTags(new HashMap<>(tags == null ? new HashMap<>() : tags));
```

Comment:
```suggestion
            alert.setTags(withCweTag(tags, cweId));
```
and let `withCweTag` add the tag to the new map if/when needed, it should be simpler than add the tag to the `tags` and then having to undo.